docs/products/cassandra/concepts/tombstones.rst

```diff
@@ -0,0 +1,62 @@
+Tombstones also result in a lot of garbage collection, which can affect cluster stability. The two main things affected are
+read performance and disk usage.
+
+**Read performance**: if read queries have to scan large numbers of tombstones, the query performance can be significantly
+degraded. In particularly bad cases, the query can even time out. There are a couple of types of queries that are more likely
+to be affected by this. They all involve scanning all or a large part of a table.
+
+* Full table scans like ``SELECT * from inventory.items``
+* Any query that requires adding ``ALLOW FILTERING``
+* Range queries, i.e. queries with ``WHERE item_cost > threshold`` or similar
+
+**Disk usage**: if you are rapidly filling up your cluster with data at the same time as you are doing a lot of
+deletions, you will reach size limits sooner. This is because the tombstone data is not actually deleted and still taking up
+space on disk.
+
+
+If you suspect problems caused by tombstones for your cluster, you can check the logs. By default, if a query encounters over
+1000 tombstones (configured by `tombstone_warn_threshold <cassandra-tombstone-warn_>`_) it will generate a log entry. The
+entry will be in the format ``Read <X> live rows and <Y> tombstone cells for query <query> [...] (see tombstone_warn_threshold)``.
+
+If it encounters over 100 000 tombstones (configured by ``tombstone_failure_threshold)``, the query will be aborted with a
+``TombstoneOverwhelmingException`` (or just time out). To investigate a query that is encountering tombstones, the easiest
+way is to connect with a ``cqlsh`` session and run ``TRACING ON`` followed by the query of interest. You can view values of
+Cassandra settings with ``SELECT * FROM system_views.settings WHERE name = '<setting name>';``.
+
+Preventing tombstone problems
+-----------------------------
+
+Often there is no "quick fix" solution for tombstone-related performance problems. Instead, you might want to rethink your
+data model or your query strategy. Some of the solutions available are:
+
+* Change your data model so that you don't have to delete so much data.
+* Avoid queries that end up running on all partitions in a table (queries with no ``WHERE`` clause, queries that need
+  ``ALLOW FILTERING``).
+* Update your queries so that they don't have to scan over tombstone rows in the same manner. For range queries, this might
+  mean investigating if you can use a narrower range, or use a different approach to the query.
+* If you are planning to delete all the data in a table, you can drop the whole table to avoid creating tombstones.
+
+Removing existing tombstones
+----------------------------
+
+The best is to wait for it to happen automatically as part of regular operations. Once more time than ``gc_grace_seconds``
+has elapsed and a compaction happens, the tombstones will be removed automatically.
+
```

Comment:
Shall we mention about a couple of commands (as a last resort with CAUTION) to be used in case of urgent need to remove expired tombstones?

1. `nodetool compact` : Forceful compaction of SSTables, it demands large free disk space and hence should be used on a limited set of selective tables at a time. If this is run on a table with STCS compaction strategy, it may cause a large SSTable.

2. `nodetool garbagecollect` : It runs a series of small compactions that also checks overlapping SSTables. While it uses less amount of disk space in the process, it is more CPU intensive and longer process execution than the `nodetool compact` option

Reply:
These are good points, but there is no way for the customer to run `nodetool` commands against their Aiven service, so I deliberately any mentions out here.

Reply:
thnx for clarifying @allhailwesttexas